library/src/G4CMPPhononElectrode.cc

```diff
@@ -188,10 +189,13 @@ ProcessReflection(const G4Track& track, const G4Step& step,
   } while (!G4CMP::PhononVelocityIsInward(theLattice, pol,
 					  reflectedKDir, surfNorm));
 
-  if (verboseLevel>1)
-    G4cout << " Phonon reflected from QET toward " << reflectedKDir << G4endl;
+  FillParticleChange(particleChange, track, reflectedKDir);
 
-  particleChange.ProposeMomentumDirection(reflectedKDir);
+  if (verboseLevel>1){
+    G4cout << " Phonon reflected from QET with: Wavevector " << reflectedKDir
+    << ", Momentum " << *particleChange.GetMomentumDirection()
```

Comment:
This should not have a pointer-dereference, and it shouldn't have compiled.  Just `particleChange.Get...`.